config/src/genesis_config.rs

```diff
@@ -718,7 +718,7 @@ static G_EMPTY_BOOT_NODES: Lazy<Vec<MultiaddrWithPeerId>> = Lazy::new(Vec::new);
 const ONE_DAY: u64 = 86400;
 
 // for test
```

Comment:
Please remove this constant and use genesis config to set dag height.  See G_DEV_CONFIG.

Reply:
OK

Reply:
Historically, this constant is used for transaction timeout and if you make sure that it is should be removed, you remove in your branch.
There are two logic in test case: one for the single chain and other for the dag chain. Both need run in the test net and 0 as fork number. But I saw the fork height in Genesis Config is u64::MAX which is not suitable for those two cases.

Reply:
And some test case, such as test_multiple_node_sync, the miner service is a must to mint blocks automatically. But in dev net, because the config set, the block will not be minted leading to failure or timeout.

Reply:
As you said, there's still lots of failed cases which need to a fix if we set `dag-fork-height` as ZERO in genesis config. I don't think this PR has done the work as it's title says, right? Please do the job like what you said.

Reply:
> As you said, there's still lots of failed cases which need to a fix if we set `dag-fork-height` as ZERO in genesis config, right? I don't think this PR has done the work as it's title says, right? Please do the job like what you said.

which case? CI is successfully.

Reply:
> Historically, this constant is used for transaction timeout and if you make sure that it is should be removed, you remove in your branch. There are two logic in test case: one for the single chain and other for the dag chain. Both need running in the test net and 0 as fork number. But I saw the fork height in Genesis Config is u64::MAX which is not suitable for those two cases.

This constant inject itself into normal function and we can configure the dag height as zero in the Test genesis config to avoid it.  At future, we will remove this function `dag_fork_height` and related codes, because we donot need it to check if if the chain is DAG or not. It should always be. Removing it is essential, not just a PR to fix the mess.

Reply:
> And some test case, such as test_multiple_node_sync, the miner service is a must to mint blocks automatically. But in dev net, because the config set, the block will not be minted leading to failure or timeout.

So change it to right value, run the tests and fix it. THX.

Reply:
Why no checking the dag or not ? 
This is the branch sync test cases using dag fork height 0 run successfully. Any other big feature or improvement should be in other branches.

Reply:
> > And some test case, such as test_multiple_node_sync, the miner service is a must to mint blocks automatically. But in dev net, because the config set, the block will not be minted leading to failure or timeout.
> 
> So change it to right value, run the tests and fix it. THX.

I really did not understand what you want. But this time, you mean I have to do the job that let the starcoin run in fork height 0 by using the config you defined? If it is, that is fine I am happy to do that. But you can say your requirement directly next time.

Reply:
> Why no checking the dag or not ? This is the branch sync test cases using dag fork height 0 run successfully. Any other big feature or improvement should be in other branches.

Yes. No need to check it, The genesis configuration make sure it always be a DAG, every blocks, headers are constructed for DAG.

Reply:
> > > And some test case, such as test_multiple_node_sync, the miner service is a must to mint blocks automatically. But in dev net, because the config set, the block will not be minted leading to failure or timeout.
> > 
> > 
> > So change it to right value, run the tests and fix it. THX.
> 
> I really did not understand what you want. But this time, you mean I have to do the job that let the starcoin run in fork height 0 by using the config you defined? If it is, that is fine I am happy to do that. But you can say your requirement directly next time.

If you donot want, don't do it. This is a small team, everyone shoud do his best.

Reply:
As the pr title shows and at the meeting I was assigned to let the test cases ci run successfullly at fork height 0 and the rest   will be done in your branch, now you ask me to do the rest? Ensure this or I am afraid of the code conflict.

Reply:
> As the pr title shows and at the meeting I was assigned to let the test cases ci run successfullly at fork height 0 and the rest will be done in your branch, now you ask me to do the rest? Ensure this or I am afraid of the code conflict.

Suit yourself.